src/main/kotlin/no/nav/tilleggsstonader/sak/hendelser/personhendelse/dødsfall/DødsfallHåndterer.kt

```diff
@@ -0,0 +1,137 @@
+            val lagretTask = taskService.save(opprettOppgaveTask)
+
+            hendelseRepository.insert(
+                Hendelse(
+                    type = TypeHendelse.PERSONHENDELSE,
+                    id = dødsfallHendelse.hendelseId,
+                    metadata = mapOf("taskId" to lagretTask.id.toString()),
+                ),
+            )
+        }
+    }
+
+    private fun finnFagsakerMedLøpendeUtbetalingFraDødsdato(dødsfallHendelse: DødsfallHendelse) =
+        fagsakService
+            .finnFagsaker(dødsfallHendelse.personidenter.toSet())
+            .filter { harAktivtVedtak(it, dødsfallHendelse.dødsdato) }
+
+    private fun List<Fagsak>.opprettTask(dødsfallHendelse: DødsfallHendelse): Task {
+        val fagsak = first()
+        logger.info("Oppretter oppgave for håndtering av dødsfall for person ${fagsak.id}")
+
+        // TODO - ved nye stønadstyper som ikke behandles av NAY bør vi lagre enhet på fagsak og opprette oppgave til alle enheter
+        return OpprettDødsfallOppgaveTask
+            .opprettTask(
+                stønadstype = fagsak.stønadstype,
+                dødsfallHendelse = dødsfallHendelse,
+            ).copy(triggerTid = LocalDateTime.now().plusWeeks(1))
+    }
+
+    private fun harAktivtVedtak(
+        fagsak: Fagsak,
+        dødsdato: LocalDate,
+    ): Boolean {
+        val behandling = behandlingService.finnSisteIverksatteBehandling(fagsak.id)
+        if (behandling != null) {
+            return vedtaksperiodeService
+                .finnVedtaksperioderForBehandling(behandling.id, null)
+                .any { it.inneholder(dødsdato) }
+        }
+        return false
+    }
+
+    private fun håndterAnnullertDødsfall(hendelseId: String) {
+        val eksisterendeHendelse = hendelseRepository.findByTypeAndId(TypeHendelse.PERSONHENDELSE, hendelseId)
+        if (eksisterendeHendelse != null) {
+            håndterAnnullertDødsfallhendelse(eksisterendeHendelse)
+        } else {
+            logger.debug("Annullert dødsfallhendelse $hendelseId")
+        }
+    }
+
+    private fun håndterAnnullertDødsfallhendelse(hendelse: Hendelse) {
+        val taskId = objectMapper.readTree(hendelse.metadata!!.json)["taskId"].first().asLong()
+        val opprinneligTask = taskService.findById(taskId)
+        if (opprinneligTask.status != Status.FERDIG) {
+            logger.info(
+                "Mottatt annullering av dødshendelse ${hendelse.id}, opprinnelig task er ikke kjørt, lar den kjøre ferdig",
+            )
+        } else {
+            val oppgave = oppgaveService.hentOppgave(opprinneligTask.metadata["oppgaveId"]!!.toString().toLong())
+            if (oppgave.erÅpen()) {
+                logger.info(
+                    "Oppgave er allerede opprettet og åpen for dødshendelse ${hendelse.id}, oppdaterer oppgave",
+                )
+                oppgaveService.oppdaterOppgave(oppgave.copy(beskrivelse = oppgave.beskrivelse + ANNULLERT_DØDSFALL_BESKRIVELSE))
```

Comment:
Oppgavene i gosys får vanligvis en ny rad med en tidspunk. Tror ikke vi har noe kode for det her, men i integrasjoner har vi en metode `lagOppgaveBeskrivelseFordeling` som viser litt hvordan det appendes. 

Samtidig trenger vi kanskje ikke det når vi bare skal informere om at dødshendelsen er annulert

Reply:
:+1: Dro inn litt kode fra integrasjoner og la i oppgaveservice ✔️